### DISCUSSION TITLE
How to use the menu

### DISCUSSION THREAD

**Discussion by Jip-Hop** [zbm-dev/zfsbootmenu#284] [Questions]
Hi! My menu doesn't look like the one in the readme:
![](https://github.com/zbm-dev/zfsbootmenu/raw/master/media/v1.11.0-multi-be.png?raw=true)

I only have `[RETURN] boot` `[CTRL-R] recovery shell` `[CTRL+H] help` in my menu. The other keyboard shortcuts are not listed. Would be helpful to see them too. How can I enable that?

I tried `[CTRL+H]` to read the help items. But how can I scroll down to read more of the text? Arrow down will move to the next item (e.g. from `Main Screen` to `ZPOOL Health` even though there's still more text to read for `Main Screen`). I tried mouse buttons, arrow keys, spacebar, enter... Can't figure out how to read the rest xD

Also, is there an option to reboot from the ZFSbootmenu directly? I get `System has not been booted with systemd as init system` error when trying reboot from ZFSbootmenu emergency shell.

**Comment by zdykstra** [comment]
`ZFSBootMenu` relies on the shell utility `column` to show the full keybind text, which is typically part of the `util-linux` package (or a variant of that name). With out it, it falls back to a limited menu keybind text. On what OS are you building the ZFSBootMenu initramfs?

You can use shift+arrow keys to scroll the help text. It sounds like a bit of additional text somewhere to explain that would be helpful.

`ctrl-alt-del` might be respected to reboot your system from inside ZFSBootMenu; let me know if that works for you.

**Reply by Jip-Hop** [reply]
> `ctrl-alt-del` might be respected to reboot your system from inside ZFSBootMenu; let me know if that works for you.

Thanks, this reboots my system :) Perhaps this could also be inside the help menu? It's great that there's a search function inside the help menu. When searching for `reboot` I didn't find anything. Shall I add `ctrl-alt-del` to the help menu?

> shift+arrow keys

This is not working for me... It switches to the next help item instead.

> On what OS are you building the ZFSBootMenu initramfs?

I followed [Debian Bullseye installation with ESP on the zpool disk](https://github.com/zbm-dev/zfsbootmenu/wiki/Debian-Bullseye-installation-with-ESP-on-the-zpool-disk).

**Comment by Jip-Hop** [comment]
I figured out how to scroll down in the help menu: https://github.com/zbm-dev/zfsbootmenu/blob/037bb55aa7923bd501cd7fa2bc020fed22e10010/zfsbootmenu/libexec/zfsbootmenu-help#L36
Using the `Page Up` and `Page Down` keys.

Would be great if a hint could be displayed for this when opening the help menu... Maybe above the hint for `[ESCAPE] back`. Also, some keyboards don't have these keys. Mine does, they are activated with the arrow up/down keys in combination with the Fn key.

**Comment by zdykstra** [comment]
I've added keybind documentation to the online help browser.

**Reply by Jip-Hop** [reply]
Thank you!

Should we document the `reboot` functionality too? I was thinking to add something like this to `docs/pod/online/recovery-shell.pod`:

```
=item I<reboot>

The reboot command isn't working. Instead you can press CTRL+ALT+DELETE to reboot. This works not just in the Recovery Shell, but anywhere in zfsbootmenu.
```

**Reply by ahesford** [reply]
No. If we document things that don't work as they do on a full system, we'd write the longest manual ever conceived. Ctrl+Alt+Del is a well known key sequence to effect a reboot, rebooting from ZFSBootMenu is not an ordinary operation, and it would also be possible to just press the reset or power button.

**Reply by Jip-Hop** [reply]
Fair enough! Just one more thing I was wondering about. The shortcuts in the Help menu are written as MOD+J. But the main menu shows CTRL+J. Why is the CTRL key called MOD in the Help menu?

**Reply by zdykstra** [reply]
The modifier key can be one of CTRL, ALT, or CTRL+ALT . Writing that out as MOD was the best compromise we could come up with.

**Reply by Jip-Hop** [reply]
Thanks again for answering all my questions. 🙂

Why is `MOD`/`CTRL` inconsistent between the main screen and the help pages? And has it been considered to add a "welcome" page to the help menu? Perhaps something like:

## Help Menu
Welcome to ZFSBootMenu Help Menu.

- Type to search inside help chapters
- Switch chapters with `Up` and `Down` arrow keys
- Scroll using `Page Up` and `Page Down` keys
- The `MOD` in keyboard shortcuts can be `CTRL`, `ALT`, or `CTRL`+`ALT`
- Press `CTRL`+`ALT`+`DELETE` to reboot
